### PR TITLE
Propose BEP003, BEP004 and withdraw BEP005

### PR DESCRIPTION
**Pull request by michaelweinold**
Dear board members (@cmutel, @romainsacchi, @bsteubing, @tngTUDOR, @haasad),

please find below some updated related to the upcoming Steering Council meeting on Wednesday.

Best,
Michael

### CODE REVIEW

File: proposals/0004_changelog-strategy.md [hunk 1]
```diff
@@ -120,4 +121,7 @@ The prototype of [documentation.brightway.dev](https://documentation.brightway.d
 
 ## Previous Versions
 
- - Version 1 (unpublished, compare PR)
\ No newline at end of file
+ - [Version 1](https://github.com/brightway-lca/enhancement-proposals/blob/2d904c1e457ad589c2e2ea7d2434c7d360989a0f/proposals/0004_changelogs.md)
+ - [Version 2](https://github.com/brightway-lca/enhancement-proposals/blob/662bd104ba8c83e313b966a9c5893b2e3c95376a/proposals/0004_changelogs.md)
+ - [Version 3](https://github.com/brightway-lca/enhancement-proposals/blob/a3043a12d1e50f27dc7c809ca50e0dd76e909381/proposals/0004_changelogs.md)
+ - Version 4 (this version)
```
Comment (cmutel) [inline proposals/0004_changelog-strategy.md:127]: This doesn't agree with the metadata section earlier.

**Comment by cmutel** [comment]
LGTM. Updating the proposals is (of course) allowed.

**Comment by michaelweinold** [comment]
Dear board members (@cmutel, @romainsacchi, @bsteubing, @tngTUDOR, @haasad, ...),

I have now updated the three BEPs I created two years ago:

BEP0003: Documentation (proposed)
BEP0004: Changelogs (proposed)
BEP0005: Dependencies (withdrawn)

**Please vote on them until the end of April by leaving a comment in this PR.**

As you can see, some of this work goes back to 2021. There is no need to have stale issues in the BEP repository. I will therefore withdraw all proposed BEPs if <50% of steeting committee members have voted by the end of April.

Many thanks!

Best,
Michael

**Comment by romainsacchi** [comment]
Unsure about BP0003: it sounds good, but seems like a lot of work
BP0004 is an easy "yes" for me. It seems like a good practice.
Not sure I understand the proposal of removing BP0005, since it's about multifunctionality.

**Comment by bsteubing** [comment]
Hi Michael, I don't know if this board is still alive or not, but I vote in favor of the choices you made for the three proposals. It sounds like good decisions and as Romain says BP0003 is certainly a lot of work...

**Comment by tngTUDOR** [comment]
I vote in favor of BEP003

**Comment by tngTUDOR** [comment]
~@michaelweinold  if you are withdrawing BEP005, please update the pull request (the readme mentions BEP005)~
Nevermind, I foresaw the "withdrawn" part in the README

File: proposals/0005_dependency-strategy.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+83. wheel
+84. wrapt
+85. xlrd
+86. xlsxwriter
+
+## Abstract
+
+New dependencies for Brightway should be either: \
+Platform independent (with support for the three major plaforms [`x86_64`, `aarch64`, `wasm32`, as defined in  PEP11](https://peps.python.org/pep-0011/)), or... \
+Fail gracefully only on first call (not on package import), according to the _"lazy import"_ logic defined in [PEP690](https://peps.python.org/pep-0690/). \
```
Comment (tngTUDOR) [inline proposals/0005_dependency-strategy.md:109]: non-blocking: It seems this pep got "rejected" upstream.

File: proposals/0003_documentation-strategy.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+|    | serve our study | serve our work |
+| -: | :-------------: | :------------: |
+practical steps | __Tutorials (=E-Book)__ <br/> Pages at <br/>  [`learn.brightway.dev`](https://github.com/brightway-lca/brightway-training) | __How-To Guides__ <br/> "Gallery" at <br/> [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation) |
+theoretical knowledge | __Explanation__ <br/> "LCA" at <br/> [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation)| __Reference__ <br/> "API Reference" at <br/>  [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation) |
+
+Table 2: The [Brightway](https://github.com/brightway-lca) implementation of the [Diataxis framework](https://diataxis.fr/)
+
+##### Reference
+
+This shall contain the API reference for the various Brightway modules (`bw_agg`, `bw_calc`, `bw_data`, ...). This is compiled automatically from the function docstrings by the [Sphinx package](https://www.sphinx-doc.org/en/master/).
```
Comment (tngTUDOR) [inline proposals/0003_documentation-strategy.md:57]: blocking: let's rephrase this to say that the reference shall contain the API reference for the "core brightway modules". They are defined elsewhere as such.

File: proposals/0003_documentation-strategy.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+### Non-Goals
+
+N/A
+
+### Test plan and results
+
+Prototypes of [docs.brightway.dev](https://docs.brightway.dev/) and [learn.brightway.dev](https://learn.brightway.dev/) were developed as part of the [Brightcon 2022](http://web.archive.org/web/20220928071701/https://2022.brightcon.link/) Hackathon. The prototype of [live.brightway.dev](https://live.brightway.dev/) was developed as part of the [Brightcon 2023](https://web.archive.org/web/20230918043912/https://2023.brightcon.link/) hackathon.
+
+The readthedocs.org traffic analysis confirms the success of the new documentation (data as per December 2023):
```
Comment (tngTUDOR) [inline proposals/0003_documentation-strategy.md:105]: non-blocking: if we have more up to date figures here, we should add them before merging.

File: proposals/0004_changelog-strategy.md [hunk 5]
```diff
@@ -32,14 +32,18 @@ Currently, there is no way for users to obtain information on feature changes an
 | `regional` | https://github.com/brightway-lca/brightway2-regional/blob/master/CHANGES.md |
 | `regional` | https://github.com/brightway-lca/bw_migrations/blob/master/CHANGELOG.md |
```
Comment (tngTUDOR) [inline proposals/0004_changelog-strategy.md:33]: typo: line # 33 says `regional` but it must be "migrations"

**Comment by tngTUDOR** [comment]
I also vote in favor of adopting BEP004 (changes)

File: proposals/0003_documentation-strategy.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+## Proposal
+
+All code added to the Brightway code base shall include docstrings conforming to the [Numpy Style Guide](https://numpydoc.readthedocs.io/en/latest/format.html). Existing code that does not conform to this standard shall be considered legacy code and shall be refactored. New code that does not conform to this standard shall not be merged into the Brightway code base.
+
+Based on the Diataxis framework, all Brightway documentation shall be structured as follows:
+
+|    | serve our study | serve our work |
+| -: | :-------------: | :------------: |
+practical steps | __Tutorials (=E-Book)__ <br/> Pages at <br/>  [`learn.brightway.dev`](https://github.com/brightway-lca/brightway-training) | __How-To Guides__ <br/> "Gallery" at <br/> [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation) |
+theoretical knowledge | __Explanation__ <br/> "LCA" at <br/> [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation)| __Reference__ <br/> "API Reference" at <br/>  [`docs.brightway.dev`](https://github.com/brightway-lca/brightway-documentation) |
```
Comment (tngTUDOR) [inline proposals/0003_documentation-strategy.md:51]: blocking: the BEP suggest in table 2 that we use "LCA" for the theory. The current main branch of the documentation does not have this anymore. I believe it's "Overview" that has replaced this. @michaelweinold  I suggest we update the BEP to say "Overview" instead of LCA.

File: proposals/0004_changelog-strategy.md [hunk 7]
```diff
@@ -87,16 +91,17 @@ Additional labels for pre-release and build metadata are available as extensions
...
-Pros: Uniform formatting and structure across the changelog files of all Brightway packages. Improved readability, improved accountability of developers. Improved accessibility by end-users. \
-Cons: None
+Pros: Uniform formatting and structure across the changelog files of all Brightway packages. Improved readability, improved accountability of developers. Improved accessibility by end-users.
+
+Cons: None (!)
 
 ### Alternatives
 
-Maintain status quo, do not include changelog information in the documentation website.
+Maintain status quo, inconsistently include changelogs in the documentation but without a common naming and formatting convention.
```
Comment (tngTUDOR) [inline proposals/0004_changelog-strategy.md:105]: non-blocking: we could use [gnu style changelogs](https://www.gnu.org/prep/standards/html_node/Style-of-Change-Logs.html)

**Review by tngTUDOR — CHANGES_REQUESTED** [review]
I have added a couple of blocking and non-blocking comments.
